### ISSUE TITLE
cmake_core.cmake does not set some options for ClangCL and improperly sets others

### ISSUE THREAD

**Issue by MarkCallow** [ARM-software/astc-encoder#415]
If the compiler in use is clang-cl many options `cmake_core.cmake` is setting will never be set and others will cause errors.

The former occurs in the section dealing with invariance because the `CLANG-LIKE` test is guarded with a `Linux,Darwin` test. If the guard is removed, you'll get compilation errors because clang-cl has the same command-line options as MSVC and does not support those of regular clang. As `CXX_COMPILER_ID` is "Clang" and the version numbering is totally different from MSVC's the lines for MSVC do nothing. Here is a suggested change:

```
--- a/lib/astc-encoder/Source/cmake_core.cmake
+++ b/lib/astc-encoder/Source/cmake_core.cmake
@@ -206,6 +206,7 @@ macro(astcenc_set_properties ASTCENC_TARGET_NAME ASTCENC_IS_VENEER)
             PRIVATE
                 $<$<CXX_COMPILER_ID:MSVC>:/fp:precise>
                 $<$<AND:$<CXX_COMPILER_ID:MSVC>,$<VERSION_GREATER_EQUAL:$<CXX_COMPILER_VERSION>,19.30>>:/fp:contract>
+                $<$<AND:$<PLATFORM_ID:Windows>,$<CXX_COMPILER_ID:${CLANG_LIKE}>>:/fp:precise>
                 $<$<AND:$<PLATFORM_ID:Linux,Darwin>,$<CXX_COMPILER_ID:${CLANG_LIKE}>>:-ffp-model=precise>
                 $<$<PLATFORM_ID:Linux,Darwin>:-ffp-contract=fast>)
     else()
@@ -215,6 +216,7 @@ macro(astcenc_set_properties ASTCENC_TARGET_NAME ASTCENC_IS_VENEER)
             PRIVATE
                 $<$<AND:$<CXX_COMPILER_ID:MSVC>,$<VERSION_LESS:$<CXX_COMPILER_VERSION>,19.30>>:/fp:strict>
                 $<$<AND:$<CXX_COMPILER_ID:MSVC>,$<VERSION_GREATER_EQUAL:$<CXX_COMPILER_VERSION>,19.30>>:/fp:precise>
+                $<$<AND:$<PLATFORM_ID:Windows>,$<CXX_COMPILER_ID:${CLANG_LIKE}>>:/fp:precise>
                 $<$<AND:$<PLATFORM_ID:Linux,Darwin>,$<CXX_COMPILER_ID:${CLANG_LIKE}>>:-ffp-model=precise>
                 $<$<PLATFORM_ID:Linux,Darwin>:-ffp-contract=off>)
```

An example of the latter occurs at line 188 to 196:
```
    if(${ASTCENC_ASAN})
        target_compile_options(${ASTCENC_TARGET_NAME}
            PRIVATE
                $<$<CXX_COMPILER_ID:${CLANG_LIKE}>:-fsanitize=address>)

        target_link_options(${ASTCENC_TARGET_NAME}
            PRIVATE
                $<$<CXX_COMPILER_ID:${CLANG_LIKE}>:-fsanitize=address>)
    endif()
```

Since clang-cl is `CLANG_LIKE` this unknown option will be set There may be others. I happened to spot this one. As I'm not using ASAN it has not affected my clang-cl builds.

**Comment by solidpixel** [comment]
> $<$<AND:$<PLATFORM_ID:Windows>,$<CXX_COMPILER_ID:${CLANG_LIKE}>>:/fp:precise>

Unfortunately this breaks normal Clang on Windows. Suspect this needs a large cleanup filtering generators based on `CXX_COMPILER_FRONTEND_VARIANT`. PR incoming shortly.

**Comment by solidpixel** [comment]
I think we need something like this PR: #417, but only lightly tested so far.

**Comment by solidpixel** [comment]
Closed - fixed in bc79f71.
